### PR TITLE
Dev/121 Problem not found page

### PR DESCRIPTION
**Pull request by Chaoyuuu**
image

### CODE REVIEW

**Review by Wally5077 — APPROVED** [review]
LGTM

**Review by nyngwang — APPROVED** [review]
LGTM.

File: src/app/problem/ProblemEditor.js [hunk 1]
```diff
@@ -14,27 +14,37 @@ import TestCase from "./edit/testCase/TestCasesList";
...
-import {useProblemContext} from "./list/ProblemContext";
+import ProblemNotFound from "./ProblemNotFound";
+import {problemService} from "../../services/services";
 
 const ProblemEditor = () => {
     const {problemId} = useParams()
-    const {currentProblem, setCurrentProblem, refetchProblem, setShouldRedirect} = useProblemContext()
-    const [problemSaved, setProblemSaved] = useState(false)
+    const [shouldRedirect, setShouldRedirect] = useState(false)
+    const [problemIdNotFound, setProblemIdNotFound] = useState(false)
```
Comment (Johnny850807) [inline src/app/problem/ProblemEditor.js:23]: `problemNotFound, setProblemNotFound`

File: src/app/problem/ProblemEditor.js [hunk 2]
```diff
@@ -75,7 +85,7 @@ const ProblemEditor = () => {
                             </section>
                             <section>
                                 <EditorButton text={"Save Change"} buttonColor={"#96D745"} fontColor={"#FFFFFF"}
-                                              onClick={() => setProblemSaved(true)}/>
+                                              onClick={onSaveProblem}/>
```
Comment (Johnny850807) [inline src/app/problem/ProblemEditor.js:88]: rename to `onProblemSaved`

File: src/app/problem/list/ProblemList.js [hunk 3]
```diff
@@ -1,55 +1,44 @@
...
-    if (!problems || (shouldRedirect && !currentProblem)) {
+    if (!problems) {
         return <Spinner/>
     }
 
     return (
-        <>{shouldRedirect ?
-            <Redirect to={`problems/${currentProblem.id}/edit`}/> : ""}
+        <>
+            {shouldRedirect ? <Redirect to={`problems/${currentProblemId}/edit`}/> : ""}
```
Comment (Johnny850807) [inline src/app/problem/list/ProblemList.js:41]: Add a comment to describe why you cannot write
```js
if (shouldRedirect) {
    return (<Redirect to={`problems/${currentProblemId}/edit`}/>);
}
```

above

File: src/app/problem/list/ProblemList.js [hunk 4]
```diff
@@ -86,18 +72,20 @@ const ProblemList = () => {
...
-                }}>
+
+            <Switch>
+                <Route path="/problems/:problemId/edit">
                     <ProblemEditor/>
-                </ProblemContext.Provider>
-            </Route>
+                </Route>
+                <Route path="/problems/:problemId/*">
+                    <ProblemNotFound/>
```
Comment (Johnny850807) [inline src/app/problem/list/ProblemList.js:86]: should redirect to `/problems/:problemId`

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done!

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Pull request merged**